inbm/dispatcher-agent/dispatcher/ota_parser.py

```diff
@@ -55,7 +55,7 @@ def parse(self, resource: Dict, kwargs: Dict, parsed: XmlHandler) -> Dict[str, A
         self._password = resource.get('password', None)
         self._signature = resource.get('signature', None)
         if self._signature:
-            self._hash_algorithm = int(resource.get('sigversion', 384))
+            self._hash_algorithm = int(resource.get('sigversion', DEFAULT_HASH_ALGORITHM))
```

Comment:
**Suggestion:** Validate the 'sigversion' value to ensure it is an integer. If it is not, raise an appropriate exception to prevent potential type errors later in the code. [error handling]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if self._signature:
            sigversion = resource.get('sigversion', str(DEFAULT_HASH_ALGORITHM))
            if sigversion.isdigit():
                self._hash_algorithm = int(sigversion)
            else:
                raise ValueError(f"Invalid 'sigversion' value: {sigversion}. It must be an integer.")
```

Reply:
out of scope of this PR